kpi/views/v2/data.py

```diff
@@ -296,6 +310,19 @@ def retrieve(self, request, pk, *args, **kwargs):
                 raise Http404
         return Response(submission)
 
+    @action(detail=True, methods=['GET'],
+            renderer_classes=[renderers.JSONRenderer],
+            permission_classes=[DuplicateSubmissionPermission])
+    def duplicate(self, request, pk, *args, **kwargs):
+        """
+        Creates a duplicate of the submission with a given `pk`
+        """
+        deployment = self._get_deployment()
+        duplicate_response = deployment.duplicate_submission(
+            requesting_user_id=request.user.id, instance_id=positive_int(pk)
+        )
+        return Response(duplicate_response)
```

Comment:
What I'd intended in #2894 but probably could've been clearer about was that the new duplication endpoint would return the same thing as `/api/v2/assets/[asset uid]/data/[new submission pk]?format=json`. That's in line with how we handle asset cloning and should allow the front end to display the new submission without making another API call. If KoBoCAT fails to return a 201 success when KPI `POST`s to its `submission` endpoint, then return a 500 with an appropriate message—if there's even anything useful to be said.

Reply:
That was my mistake in not reading your requirement correctly 🤦